hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/shell/TestOzoneShellHA.java

```diff
@@ -910,6 +910,27 @@ public void testCreateBucketWithECReplicationConfig() throws Exception {
     }
   }
 
+  @Test
+  public void testPutKeyOnBucketWithECReplicationConfig() throws Exception {
+    final String volumeName = "volume1000";
+    getVolume(volumeName);
+    String[] args =
+        new String[] {"bucket", "create", "/volume1000/bucket0", "-t", "EC",
+            "-r", "rs-3-2-1024k"};
+    execute(ozoneShell, args);
+
+    args = new String[] {"key", "put", "/volume1000/bucket0/newECKey1",
+        testFilePathString};
+    execute(ozoneShell, args);
+    OzoneVolume volume =
+        cluster.getClient().getObjectStore().getVolume(volumeName);
+    OzoneBucket bucket = volume.getBucket("bucket0");
+    try (OzoneOutputStream out = bucket.createKey("newECKey1", 1024)) {
```

Comment:
Shouldn't we open the key we have created by executing the shell command to run putkey earlier instead of creating it? (Also that would mean we get an OzoneInputStream here and we can check its type.)

Reply:
You are right. I supposed to reopen key and check type. I did that correction now. Thanks